baremetal/metal3machine_manager.go

```diff
@@ -149,6 +151,43 @@ func (m *MachineManager) role() string {
 	return bmRoleNode
 }
 
+// SetPauseAnnotation sets the pause annotations on associated bmh
+func (m *MachineManager) SetPauseAnnotation(ctx context.Context) error {
+	// look for associated BMH
+	host, err := m.getHost(ctx)
+	if err != nil {
+		m.setError("Failed to get a BaremetalHost for the Metal3Machine",
+			capierrors.CreateMachineError,
+		)
+		return err
+	}
+	if host == nil {
+		m.Log.Info("BaremetalHost not associated")
+		return nil
+	}
+
+	annotations := host.GetAnnotations()
+
+	if annotations != nil {
+		if _, ok := annotations[bmh.PausedAnnotation]; ok {
+			if host.Annotations[bmh.PausedAnnotation] != pausedAnnotationKey {
+				m.Log.Info("BaremetalHost is already paused by user")
```

Comment:
maybe this does not need to be logged, because it will jam the logs, every 30 sec. for all bmhs...

Reply:
I would rather keep this one.